### PR TITLE
refac: Settlement domain test to BDD style

### PR DESCRIPTION
**Pull request by energinet-ajw**
Rewritten download settlement report domain test to follow a kind of BDD.
Note that this test was previously skipped due to timeout. 

https://app.zenhub.com/workspaces/mandalorian-5fd22a1a59e4740018f5ab5a/issues/gh/energinet-datahub/team-mandalorian/80

### CODE REVIEW

File: source/dotnet/domain-tests/Fixtures/SettlementReportScenarioFixture.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                settlementDownloadInput.ProcessType,
+                settlementDownloadInput.CalculationPeriodStart,
+                settlementDownloadInput.CalculationPeriodEnd);
+            DiagnosticMessageSink.WriteDiagnosticMessage($"Downloading settlement report for " +
+                                                         $"grid area codes {string.Join(", ", settlementDownloadInput.GridAreaCodes.ToArray())} and" +
+                                                         $" process type {settlementDownloadInput.ProcessType} started.");
+            return fileResponse;
+        }
+
+        public async Task<string[]> SplitEntryIntoLines(ZipArchiveEntry entry)
```
Comment (dstenroejl) [inline source/dotnet/domain-tests/Fixtures/SettlementReportScenarioFixture.cs:55]:
```suggestion
        public async Task<string[]> SplitEntryIntoLinesAsync(ZipArchiveEntry entry)
```

File: source/dotnet/domain-tests/SettlementFeatureTests.cs [hunk 2]
```diff
@@ -26,46 +26,64 @@ namespace Energinet.DataHub.Wholesale.DomainTests
...
+                // Assert
+                Fixture.ScenarioState.CompressedSettlementReport.Entries.Should().NotBeEmpty();
+            }
+
+            [Priority(3)]
+            [DomainFact]
+            public void AndThen_FirstEntryNameShouldBeResultCsv()
+            {
+                var expected = "Result.csv";
+                Fixture.ScenarioState.Entry = Fixture.ScenarioState.CompressedSettlementReport.Entries.Single();
```
Comment (dstenroejl) [inline source/dotnet/domain-tests/SettlementFeatureTests.cs:71]: Here we call "Single()" but does this mean we expect only one result, or do we mean "First()"?
The method is named "First" so I'm just not sure if we mean "First" or "Single".

File: source/dotnet/domain-tests/SettlementFeatureTests.cs [hunk 3]
```diff
@@ -26,46 +26,64 @@ namespace Energinet.DataHub.Wholesale.DomainTests
...
 
-                var lines = content.Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries);
+                // Assert
+                Fixture.ScenarioState.Entry.Name.Should().Be(expected);
+            }
 
-                foreach (var line in lines[1..])
+            [Priority(4)]
+            [DomainFact]
+            public async Task AndThen_FirstEntryShouldContainCorrectGridAreaCodesAndProcessType()
```
Comment (dstenroejl) [inline source/dotnet/domain-tests/SettlementFeatureTests.cs:79]: Is it important that we reference to "FirstEntry", then maybe we should call the property that in the "state" object(?)

**Review by dstenroejl — COMMENTED** [review]
There was a method that should be suffixed with "Async", and some questions.

File: source/dotnet/domain-tests/SettlementFeatureTests.cs (hunk 3 above)
Comment (energinet-ajw) [reply source/dotnet/domain-tests/SettlementFeatureTests.cs]: Changed to single.

File: source/dotnet/domain-tests/SettlementFeatureTests.cs (hunk 2 above)
Comment (energinet-ajw) [reply source/dotnet/domain-tests/SettlementFeatureTests.cs:71]: Changed to single.

**Review by dstenroejl — APPROVED** [review]

**Pull request merged**